### ISSUE TITLE
Error using generate with transformer_heads model having multiple heads (LM and linear)

### ISSUE THREAD

**Issue by zthsk** [center-for-humans-and-machines/transformer-heads#19]
I'm encountering an error when calling the generate function on a model loaded with transformer_heads. My model configuration includes multiple heads: a language modeling (LM) head for text generation and several other linear heads. The error seems to arise during the forward pass when attempting to move the loss tensor to the device, specifically at:

loss = loss.to(next(iter(loss_by_head.values())).device)

Steps to Reproduce

    Load model parameters and instantiate the model:

```
model_params = get_model_params(model_path)
model_class = model_params["model_class"]        
model = load_lora_with_heads(
    model_class,
    checkpoint_path,
    quantization_config,
    device_map={"": torch.cuda.current_device()}
)
```

Generate text using the LM head:

```
output = model.generate(
    input_ids=inputs.input_ids,
    attention_mask=inputs.attention_mask,
    max_length=1024,
    do_sample=True,
    temperature=0.9,
    top_p=0.8,
    eos_token_id=tokenizer.eos_token_id
)
```

Error encountered:

The traceback indicates the issue arises during the loss computation in the forward pass:

    File ".../transformer_heads/model/model.py", line 237, in forward
        loss = loss.to(next(iter(loss_by_head.values())).device)

Questions/Clarifications Needed

    Usage of generate: Is the generate function intended to work directly with models that have multiple heads (i.e., a LM head and additional linear heads)? If so, what is the recommended approach or necessary configuration?
    Configuration Details: Should there be any specific configuration or flag to indicate that text generation is being performed (thus only using the LM head) while ignoring the other linear heads?
    Workaround: Are there any known workarounds to bypass this error when using generate with a multi-head configuration?

Any guidance on how to correctly use the generate function in this scenario would be appreciated.

**Comment by yannikkellerde** [comment]
Yeah. Probably my way of implementing generate with transformer-heads was not the smartest as it breaks with each change the huggingface transformers team makes to their function API.

Anyway, I fixed generate for now with 35d62bade771bef16ae2e08365259cf3306e40e4. I tried test_sample_generation from transformer_heads/tests/test_generate.py using transformers==4.47.1 and it works. Maybe you can work of that.

I do not think I will have time to maintain generate forever if transformers keeps changing their API.
